src/components/diagrams/diagram-pane.js

```diff
@@ -130,83 +138,134 @@ const useDisplayView = (network, studyUuid, currentNode) => {
+                    const countryName = svg.additionalMetadata.countryName;
+                    if (countryName) {
+                        label += ' - ' + countryName;
+                    }
+                    return {
+                        id: id,
+                        nodeId: currentNode.id,
+                        state: state,
+                        name: label,
+                        fetchSvg: () => fetchSvgData(svgUrl),
+                        svgType: DiagramType.SUBSTATION,
+                        ...svg,
+                    };
+                });
             }
 
             function createVoltageLevelDiagramView(id, state) {
-                const voltageLevel = network.getVoltageLevel(id);
-                if (!voltageLevel) return;
-                let label = getNameOrId(voltageLevel);
-                const substation = network.getSubstation(
-                    voltageLevel.substationId
-                );
-                const countryName = substation?.countryName;
-                if (countryName) {
-                    label += ' - ' + countryName;
-                }
                 const svgUrl = checkAndGetVoltageLevelSingleLineDiagramUrl(id);
-
-                return {
-                    id: id,
-                    ref: React.createRef(),
-                    state: state,
-                    name: label,
-                    svgUrl: svgUrl,
-                    svgType: DiagramType.VOLTAGE_LEVEL,
-                    substationId: substation?.id,
-                };
+                return fetchSvgData(svgUrl).then((svg) => {
+                    let label = getNameOrId(svg.additionalMetadata);
+                    let substationId = svg.additionalMetadata.substationId;
+                    const countryName = svg.additionalMetadata.countryName;
+                    if (countryName) {
+                        label += ' - ' + countryName;
+                    }
+                    return {
+                        id: id,
+                        nodeId: currentNode.id,
+                        state: state,
+                        name: label,
+                        fetchSvg: () => fetchSvgData(svgUrl),
+                        svgType: DiagramType.VOLTAGE_LEVEL,
+                        substationId: substationId,
+                        ...svg,
+                    };
+                });
             }
 
             function createNetworkAreaDiagramView(ids, state, depth = 0) {
-                let displayedVoltageLevels = [];
-                let nadTitle = '';
-                if (ids) {
-                    ids.forEach((id) =>
-                        displayedVoltageLevels.push(network.getVoltageLevel(id))
-                    );
+                if (ids?.length) {
+                    const svgUrl = checkAndGetNetworkAreaDiagramUrl(ids, depth);
+                    return fetchSvgData(svgUrl).then((svg) => {
+                        let nadTitle = '';
```

Comment:
The nadTitle could/should be crafted in its own function

Reply:
I just kept the previous code